### PR TITLE
CVE-2015-1293 & CVE-2010-3250

### PR DESCRIPTION
**Pull request by bbchristians**
Completed 2 CVEs for review. (See other comment on this pull request for a detail on an issue I was facing with submission)

### CODE REVIEW

**Comment by bbchristians** [comment]
Sorry about this, but I was running into some issues regarding some merge conflicts which were preventing the CI from being able to run. I am unable to include the `description` and `curated` field values in my submission, so I've included them below.

### CVE-2015-1293
curated: true
description: |
A vulnerability in a DOM implementation in web browser engine lead to remote attackers bypassing the Same Origin Policy through constructing a forged security credentials.

### CVE-2010-3250
curated: true
description: |
A vulnerability allows remote attackers access to a private list of functional plug-ins added to the application by the user.

File: cves/CVE-2010-3250.yml [hunk 1]
```diff
@@ -108,12 +137,15 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: |
+    The fixing commit was the user's first commit to the project, but has 96 other commits to the system.
+    This means that he was probably and employee and not contracted. This means that this issue was probably
+    given to the developer as a low risk/priority assignment soon after they came onto the project. This 
+    shows that the chromium team knew about the vulnerability (which was in the system for just under 2 years),
+    was a low priority fix.
```
Comment (cmb9400) [inline cves/CVE-2010-3250.yml]: Nice find! Looks like from [here](https://bugs.chromium.org/p/chromium/issues/detail?id=45876) the team that was supposed to fix this bug didn't have time so they got someone from the "security team" to fix it before a deadline

File: cves/CVE-2015-1293.yml [hunk 2]
```diff
@@ -1,6 +1,23 @@
 ---
 CVE: CVE-2015-1293
+CWE_instructions: |
+  Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
+  that describes your vulnerability. (Tip: this may not be a good one to start
+  with - spend time understanding this vulnerability before making your choice!)
 CWE:
```
Comment (cmb9400) [inline cves/CVE-2015-1293.yml]: Need a CWE here too

File: cves/CVE-2015-1293.yml [hunk 3]
```diff
@@ -27,14 +48,16 @@ reviews:
...
+repo: https://chromium.googlesource.com/v8/v8/
 fixes_vcc_instructions: |
   Please put the commit hash in "commit" below (see my example in
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
-- :commit: 3134d8a254ebda12ac2972283f724201c2fa326d
-  :note: ''
-vccs: []
+- commit: 3134d8a254ebda12ac2972283f724201c2fa326d
+  note: Changed a session terminal condition, and added new JS to display messages.
```
Comment (cmb9400) [inline cves/CVE-2015-1293.yml]: I think the note you wrote here is a little misleading, those appear to be changes related to the tests that were added for this vulnerability -- it looks to me like the fix is changing `callingWindow.frame()->securityContext()` to `callingWindow.document()`

File: cves/CVE-2010-3250.yml [hunk 4]
```diff
@@ -28,14 +48,16 @@ reviews:
 - 3116011
 bugs:
 - 45876
-repo: 
+repo: https://chromium.googlesource.com/v8/v8/
```
[the remark this answers is not in the document]
Comment (bbchristians) [reply cves/CVE-2010-3250.yml]: Updated.

File: cves/CVE-2010-3250.yml (hunk 1 above)
Comment (bbchristians) [reply cves/CVE-2010-3250.yml]: Thanks!

File: cves/CVE-2015-1293.yml (hunk 2 above)
Comment (bbchristians) [reply cves/CVE-2015-1293.yml]: Updated.

File: cves/CVE-2015-1293.yml (hunk 3 above)
Comment (bbchristians) [reply cves/CVE-2015-1293.yml]: Yep, didn't see that that first block was testing related. I updated to commit note to emphasize the importance of the insecure access terminal function.

File: cves/my_cves/desktop.ini [hunk 5]
```diff
@@ -0,0 +1,4 @@
+[LocalizedFileNames]
+CVE-2016-5128.yml=@CVE-2016-5128.yml,0
+CVE-2015-1293.yml=@CVE-2015-1293.yml,0
+CVE-2010-3250.yml=@CVE-2010-3250.yml,0
```
Comment (andymeneely) [inline cves/my_cves/desktop.ini]: You'll need to remove this file addition - should be just the yaml file.

File: cves/CVE-2015-1293.yml [hunk 6]
```diff
@@ -32,7 +32,9 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
```
Comment (MrM21632) [inline cves/CVE-2015-1293.yml]: If you haven't gotten to this yet, you should probably work on the tense and word usage in this description.

File: cves/CVE-2015-1293.yml [hunk 7]
```diff
@@ -94,20 +101,25 @@ discovered:
...
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    A user reported this bug after running personal tests. The user supplied the Chromium team with
+    a use case, and the broken function. This vulnerability was discovered by ' marius.mlynski@gmail.com'
```
Comment (MrM21632) [inline cves/CVE-2015-1293.yml]: I noticed that this was the same researcher who found one of the CVEs I worked on (CVE-2016-1711). I think the developer who worked on this vulnerability was also the one who worked on CVE-2016-1711. What's even more interesting is that it deals with the same issue that CVE-2016-1711 revolves around (i.e., bypassing the Same Origin Policy in Blink). Seems like they were dealing with this issue for a while.

File: cves/CVE-2015-1293.yml [hunk 8]
```diff
@@ -194,4 +227,12 @@ mistakes:
...
-  answer: 
+  answer: |
+    This vulnerability arose from an oversight by a developer. Their implementation was not 
+    secure by default which provided an easy means for potential attackers to exploit the system.
+    This vulnerability was unnoticed during code reviews, even though it was relatively 
+    obvious in hindsight. The developer who was responsible for creating the vulnerability, and
+    for implementing a resolution for it was focused on patching other vulnerabilities around the
+    time of the fix. However, the developer spent a large amount of time implementing efficiency
+    improvements before the fix was committed. This means that, had it not been found by an external
+    user, it would have likely made it deep into production.
```
Comment (MrM21632) [inline cves/CVE-2015-1293.yml]: Very interesting CVE overall. I'll give this one three "points."

File: cves/CVE-2010-3250.yml [hunk 9]
```diff
@@ -77,9 +80,12 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    It does not appear that testing was does autonomously for this section. The unit
```
Comment (MrM21632) [inline cves/CVE-2010-3250.yml]: Small word usage error here ("does" instead of "done").

File: cves/CVE-2010-3250.yml [hunk 10]
```diff
@@ -195,4 +209,12 @@ mistakes:
...
-  answer: 
+  answer: |
+    This looks like a classic example of software that had a minor vulnerability
+    most likely because the developers not wanting to put the time into fixing it.
+    Some evidence: The VCC was made the day before Christmas (near the end of the 
+    west-coast workday). The vulnerability was fixed by the first commit a contributor
+    made to the system. This is highly evident that it was some minor but annoying 
+    work given to a new hire. The fix required a new development to be made, as well
+    as minor other changes in different areas of the subsystem. The vulnerability
+    existed within the system for just under 2 years.
```
Comment (MrM21632) [inline cves/CVE-2010-3250.yml]: Good work on this CVE. Pretty interesting overall, and kinda funny, too. I'll give this one two "points."

File: cves/CVE-2010-3250.yml (hunk 9 above)
Comment (bbchristians) [reply cves/CVE-2010-3250.yml]: Good catch -- updated.

File: cves/CVE-2010-3250.yml (hunk 10 above)
Comment (bbchristians) [reply cves/CVE-2010-3250.yml]: Your upvotes have been accounted for.

File: cves/CVE-2015-1293.yml (hunk 6 above)
Comment (bbchristians) [reply cves/CVE-2015-1293.yml]: I see what you mean about this, I've changed the end of the sentence to hopefully correct the issue.

File: cves/CVE-2015-1293.yml (hunk 7 above)
Comment (bbchristians) [reply cves/CVE-2015-1293.yml]: Yeah my other assigned CVE was similar as well. There was next to nothing on that one, so I chose not to do it.

File: cves/CVE-2015-1293.yml (hunk 8 above)
Comment (bbchristians) [reply cves/CVE-2015-1293.yml]: Upvotes accounted for.

File: cves/CVE-2015-1293.yml [hunk 11]
```diff
@@ -116,10 +128,20 @@ interesting_commits:
     interesting in light of the lessons learned from this vulnerability. Any
     emerging themes?
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit: 5f5d4a2e72bceee8354538d625a501822ac51b5f
+    note: |
+      A lot of commits by the VCC and Fix's developer involved other security-related
```
[the remark this answers is not in the document]
Comment (bbchristians) [reply cves/CVE-2015-1293.yml]: Thanks!

File: cves/CVE-2010-3250.yml [hunk 12]
```diff
@@ -49,22 +50,24 @@ reviews:
...
+  :note: Initial implementation of the extension protocol.
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes: 2
```
Comment (cmb9400) [inline cves/CVE-2010-3250.yml]: +1 upvote!

Comment (bbchristians) [reply cves/CVE-2010-3250.yml]: It's been accounted for.

**Pull request merged**